apps/prairielearn/src/pages/administratorInstitutions/administratorInstitutions.ts

```diff
@@ -11,7 +12,26 @@ router.get(
   '/',
   asyncHandler(async (req, res) => {
     const institutions = await sqldb.queryRows(sql.select_institutions, InstitutionRowSchema);
-    res.send(AdministratorInstitutions({ institutions, resLocals: res.locals }));
+    const availableTimezones = await getAvailableTimezonesByName();
+    res.send(
+      AdministratorInstitutions({ institutions, availableTimezones, resLocals: res.locals }),
+    );
+  }),
+);
+
+router.post(
+  '/',
+  asyncHandler(async (req, res) => {
+    if (req.body.__action === 'add_institution') {
+      await sqldb.queryAsync(sql.add_institution, {
+        short_name: req.body.short_name,
+        long_name: req.body.long_name,
+        display_timezone: req.body.display_timezone,
+        uid_regexp: req.body.uid_regexp,
```

Comment:
Nit: I think we'd rather end up with `null` than an empty string or string consisting of only whitespace:

```suggestion
        uid_regexp: req.body.uid_regexp.trim() || null,
```

Reply:
Updated this locally in commit [ed4f9f4](https://github.com/PrairieLearn/PrairieLearn/pull/9045/commits/ed4f9f44057278fce12a5e2de1948446144ae6c6).